transport/EndpointFactoryImpl.h

```diff
@@ -26,11 +31,38 @@ class EndpointFactoryImpl : public EndpointFactory
         RtcePoll& epoll,
         bool isShared);
 
-    virtual UdpEndpoint* createUdpEndpoint(jobmanager::JobManager& jobManager,
+    UdpEndpoint* createUdpEndpoint(jobmanager::JobManager& jobManager,
```

Comment:
extend EndpointFactory to handle all endpoint creation. This allows testing with emulation of other endpoints than UDP.